src/main/java/seedu/address/ui/BookCard.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.ui;
+
+import java.util.Comparator;
+import java.util.stream.Collectors;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.book.Author;
+import seedu.address.model.book.Book;
+
+/**
+ * An UI component that displays information of a {@code Person}.
+ */
+public class BookCard extends UiPart<Region> {
+
+    private static final String FXML = "BookListCard.fxml";
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
+
+    public final Book book;
+
+    @FXML
+    private HBox cardPane;
+    @FXML
+    private Label name;
+    @FXML
+    private Label id;
+    @FXML
+    private Label authors;
+    @FXML
+    private Label isbn;
+    @FXML
+    private FlowPane bookCategoryTags;
+    @FXML
+    private FlowPane bookAvailableTag;
+    @FXML
+    private FlowPane bookBorrowTag;
```

Comment:
Is it necessary to have two FlowPane since they are mutually exclusive? Only one will be shown at a time. How do you enforce that no two FlowPane will be shown at any given time?